libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4

```diff
@@ -299,9 +299,13 @@ predicate[ParserRuleContext value]
     | IS NOT? DISTINCT FROM right=valueExpression                                    #distinctFrom
     ;
 
+
+
 valueExpression
     : primaryExpression                                                              #valueExpressionDefault
     | operator=(MINUS | PLUS) valueExpression                                        #arithmeticUnary
+    | left=valueExpression operator=CARET
```

Comment:
Why not removing this line and add `CARET` to the rule below, together with `ASTERISK`, `SLASH`, and `PERCENT`?

Reply:
Hi  I cannot do that because that assigns same priority to ^  , as * / % which is not correct 

Eg 5*5^7 = 5*(5^7) = 390625. Correct when caret is above * / %
else it would give 25^7 which is incorrect as verified by official postgresql

![Screenshot 2024-03-14 at 2 00 46 AM](https://github.com/crate/crate/assets/71836462/2453c805-29c0-4223-b07d-99da9fcc868b)

Reply:
yes, correct, power has higher precedence, so we need to keep it as is

Reply:
Thx for the clarification! let's add a comment to the grammar so it's visible why the order there is important.

Reply:
I added a comment